sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/CloseReason.cs

```diff
@@ -21,7 +21,7 @@ public enum PartitionProcessorCloseReason
         /// <summary>A non-retriable exception was thrown by the Event Hub Client.</summary>
         EventHubException,
 
-        /// <summary>A non-retriable exception was thrown by the provided partition processor.</summary>
-        PartitionProcessorException
+        /// <summary>A non-retriable exception was thrown by the provided event processing handler.</summary>
```

Comment:
Not sure that this is true;  We're explicitly not handling exceptions that occur in developer-provided code.

Reply:
I agree that's what we're trying to achieve, but the current `Pump` explicitly catches any exception thrown by `ProcessEventAsync`, and this `CloseReason` is used for this kind of situation. I'll revisit this when updating error handling through code.